### PR TITLE
Branch-Add-Seller

### PR DESCRIPTION
**Pull request by KhoonSun47**
- Add necessary class for Seller functionality to work
- Update test cases
- Update check style

Points to take note:
- Can only add Condo and HDB, did not try Landed yet
because House do not have a validation for this
- Can only add one House, have not tested multiple houses
because does not have addHouse command yet
- EditCommand test cases have been commented out
because have not update EditCommand yet
- Can only add seller, need to update JsonSerializableAddressBook
to add buyer
- Need to take note of adding duplicates if buyer and seller [] were
to be seperated

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddSellerCommand.java [hunk 1]
```diff
@@ -26,31 +30,39 @@ public class AddSellerCommand extends Command {
...
+            + PREFIX_UNITNUMBER + "25 "
             + PREFIX_POSTALCODE + "578578 "
             + PREFIX_TAG + "friends "
-            + PREFIX_TAG + "owesMoney";
+            + PREFIX_TAG + "owesMoney ";
 
-    public static final String MESSAGE_SUCCESS = "New seller added: %1$s";
-    public static final String MESSAGE_DUPLICATE_PERSON = "This seller already exists in the address book";
+    public static final String MESSAGE_SUCCESS = "New seller added= %1$s";
+    public static final String MESSAGE_DUPLICATE_SELLER = "This seller already exists in the address book";
```
Comment (zengzihui) [inline src/main/java/seedu/address/logic/commands/AddSellerCommand.java:54]: Need to change the "in the address book" to "in EstateEase"

File: src/main/java/seedu/address/model/house/NonLanded.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    /**
+     * Returns a string representation of the non-landed house.
+     *
+     * @return A string representation of the non-landed house.
+     */
+    @Override
+    public String toString() {
+        String blockLevelInfo = (block != null ? "Block: " + block.value + ", " : "")
+                + (level != null ? "Level: " + level.value + ", " : "");
+        return "Non-Landed House: " + blockLevelInfo + super.toString();
```
Comment (zengzihui) [inline src/main/java/seedu/address/model/house/NonLanded.java:75]: the "Non-Landed House: " string should changed to "HDB" / "Condominium" through a checked value or something, can make changes for this part in next iteration v1.3 instead.

File: src/main/java/seedu/address/model/person/Person.java [hunk 3]
```diff
@@ -21,22 +20,21 @@ public class Person {
     private final Name name;
     private final Phone phone;
     private final Email email;
-
-    // Data fields
-    private final Address address;
-    private final PostalCode postalCode;
+    // Critically, this refers to what type the buyer WANTS, and what type the seller HAS
+    // I think this one we have to clarify whether housingType should be under House or Person,
+    // because it seems very weird to be under Person
```
Comment (zengzihui) [inline src/main/java/seedu/address/model/person/Person.java:25]: Agree, should change this in v1.3

File: src/main/java/seedu/address/storage/JsonAdaptedHouse.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+/**
+ * Jackson-friendly version of {@link House}
+ */
+public class JsonAdaptedHouse {
+    private final String block;
+    private final String level;
+    private final String postalCode;
+    private final String street;
+    private final String unitNumber;
+    private final String type; // "NonLanded" or "Landed": This is different from housingType.
```
Comment (zengzihui) [inline src/main/java/seedu/address/storage/JsonAdaptedHouse.java:25]: umm, shall we store them as Landed/HDB/Condominium instead of Nonlanded?

Comment (zengzihui) [reply src/main/java/seedu/address/storage/JsonAdaptedHouse.java:25]: for next iteration

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 5]
```diff
@@ -11,24 +11,34 @@
...
  */
 @JsonRootName(value = "addressbook")
 class JsonSerializableAddressBook {
-
     public static final String MESSAGE_DUPLICATE_PERSON = "Persons list contains duplicate person(s).";
-
-    private final List<JsonAdaptedPerson> persons = new ArrayList<>();
+    private final List<JsonAdaptedSeller> sellers = new ArrayList<>();
+    // Maybe this can be changed to buyer, I am not too sure whether should we separate the sellers list and
+    // buyer list, or keep everything under a Person?
```
Comment (zengzihui) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:24]: maybe can keep both sellers & buyers under persons in Json?

File: src/main/java/seedu/address/storage/JsonAdaptedHouse.java (hunk 4 above)
Comment (KhoonSun47) [reply src/main/java/seedu/address/storage/JsonAdaptedHouse.java:25]: Yes, we should, that one is actually meant to be HDB/Condo/Landed, but now the housingType is under House, which we need to take it out and replace type with HousingType

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java (hunk 5 above)
Comment (KhoonSun47) [reply src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:24]: Yeap but I was thinking to seperate both buyer and seller could be better for clarity, but it really depends on what you guys want for this.

File: src/main/java/seedu/address/logic/commands/AddSellerCommand.java (hunk 1 above)
Comment (KhoonSun47) [reply src/main/java/seedu/address/logic/commands/AddSellerCommand.java:54]: Okay, will do so :)

**Review by lokidoki102 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/storage/JsonAdaptedSeller.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import seedu.address.model.person.Seller;
+
+/**
+ * Jackson-friendly version of {@link Seller}.
+ */
+public class JsonAdaptedSeller extends JsonAdaptedPerson {
+    private final List<JsonAdaptedHouse> houses = new ArrayList<>();
+
+    /**
+     * Constructs a {@code JsonAdaptedSeller}, extends from JsonAdaptedHouse.
```
[the remark this answers is not in the document]
Comment (KhoonSun47) [reply src/main/java/seedu/address/storage/JsonAdaptedSeller.java:23]: JsonAdaptedPerson, thanks for pointing it out, I shall make the change

**Pull request merged**